src/main/java/duke/command/Command.java

```diff
@@ -0,0 +1,17 @@
+package duke.command;
+
+import duke.Response;
+import duke.exception.DukeException;
+/**
+ * Command represents a command to be executed.
+ */
+public abstract class Command {
+
+    /**
+     * Performs an action in response to the command
+     *
+     * @return The Response to be displayed.
+     * @throws DukeException If the action cannot be executed.
+     */
+    public abstract Response action() throws DukeException;
```

Comment:
Since method names should be verbs, perhaps you can consider renaming this method to something like `execute`.